bde/bde.py

```diff
@@ -592,6 +752,40 @@ def predict(
         # actually lists quantiles not the intervals
         raw: bool = False,
     ):
+        """Predict regression targets with optional uncertainty summaries.
+
+        Parameters
+        ----------
+        x : ArrayLike
+            Feature matrix of shape (n_samples, n_features).
+        mean_and_std : bool, default=False
+            When ``True``, return a tuple ``(mean, std)`` where both arrays
+            have shape ``(n_samples,)`` and ``std`` combines aleatoric and
+            epistemic uncertainty.
+        credible_intervals : list[float] | None, default=None
+            Quantile levels in (0, 1) used to summarise the predictive
+            distribution. When not ``None`` and ``mean_and_std`` is ``False``, the
+            method returns ``(mean, quantiles)`` where ``mean`` has shape
+            ``(n_samples,)`` and ``quantiles`` has shape ``(Q, n_samples)`` with
+            one row per requested level. Note that the values are quantiles,
+            not closed intervals; for an 80% interval you would typically pass
+            ``[0.1, 0.9]`` and interpret the two returned quantile curves as
+            lower and upper bounds.
+        raw : bool, default=False
+            When ``True``, ignore other flags and return the raw ensemble
+            outputs with shape ``(n_members, n_samples_draws, n_samples, 2)``,
+            corresponding to per-member, per-draw mean and scale parameters.
```

Comment:
add a note here & somewhere in the docs that softplus is used i.e. formalize the how the Gaussian likelihood is modeled in this framework! Might be a good fit for the readme as well